sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2862,6 +2840,8 @@ static void mergeImageData(const std::vector<device_image_plain> &Imgs,
                            device_image_impl::SpecConstMapT &NewSpecConstMap) {
   for (const device_image_plain &Img : Imgs) {
     std::shared_ptr<device_image_impl> DeviceImageImpl = getSyclObjImpl(Img);
+    if (!DeviceImageImpl->get_kernel_ids_ptr())
+      continue;
```

Comment:
There are other uses of `device_image_impl::get_kernel_ids_ptr()` which don't seem to expect it being uninitialised. Maybe it's safer to set up an empty vector in `m_BinImg2KernelIDs` as part of the device lib special handling?